src/main/java/org/openstreetmap/atlas/checks/validation/points/AddressPointMatchCheck.java

```diff
@@ -0,0 +1,136 @@
+    public static final String NO_STREET_NAME_EDGE_INSTRUCTIONS = "This node, {0,number,#}, has "
+            + "no street name specified in the address. The street name should likely "
+            + "be one of {1}. These names were derived from nearby Edges.";
+    public static final String NO_SUGGESTED_NAMES_INSTRUCTIONS = "This node, {0,number,#}, has "
+            + "no street name specified in the address. No suggestions names were found.";
+    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays.asList(
+            NO_STREET_NAME_POINT_INSTRUCTIONS, NO_STREET_NAME_EDGE_INSTRUCTIONS,
+            NO_SUGGESTED_NAMES_INSTRUCTIONS);
+    private static final String ADDRESS_STREET_NUMBER_KEY = AddressHousenumberTag.KEY;
+    private static final String POINT_STREET_NAME_KEY = AddressStreetTag.KEY;
+    private static final String EDGE_STREET_NAME_KEY = NameTag.KEY;
+    private static final double BOUNDS_SIZE_DEFAULT = 150.0;
+
+    private final Distance boundsSize;
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    public AddressPointMatchCheck(final Configuration configuration)
+    {
+        super(configuration);
+        this.boundsSize = Distance.meters(
+                (Double) configurationValue(configuration, "bounds.size", BOUNDS_SIZE_DEFAULT));
+    }
+
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        // Object is an instance of Point
+        return object instanceof Point
+                // And has a street number specified
+                && object.getTag(ADDRESS_STREET_NUMBER_KEY).isPresent()
+                // And if the street name key has a value of null
+                && (!object.getTag(POINT_STREET_NAME_KEY).isPresent()
+                        // Or if the street name key is not present
+                        || !object.getTags().containsKey(POINT_STREET_NAME_KEY));
+    }
+
+    /**
+     * This is the actual function that will check to see whether the object needs to be flagged.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object that
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final Point point = (Point) object;
+        // Get a bounding box around the Point of interest
+        final Rectangle box = point.getLocation().boxAround(boundsSize);
+
+        // Get all Points inside the bounding box
+        final Iterable<Point> interiorPoints = point.getAtlas().pointsWithin(box);
+        // Convert the Iterable into a Stream for filtering, remove Points that have null as their
+        // street name as they cannot be candidates for a street for the Point of interest
+        final Set<Point> points = StreamSupport.stream(interiorPoints.spliterator(), false)
```

Comment:
You can say `Iterable.stream(point.getAtlas().pointsWithin(box))` to stream points within an area.

Reply:
@mertemin Is there a nice way to collect an Iterables object into a Set to retain unique street names? Also, is there a concise way of checking whether Iterables are empty? I know Guava has this functionality but in straight Java, I don't think there's a pretty way of checking/doing those things.